test/toolbox_image.sh

```
@@ -57,11 +66,17 @@ ________________________________________________________________________________
 
     LOOP_STORAGE_DEVICE="$(losetup --show --find --partscan "${rootfs_dir}${STORAGE_DEVICE_IMG}")"
 
-    mkfs.ext4 -q "${LOOP_STORAGE_DEVICE}p1" && fsck -y "${LOOP_STORAGE_DEVICE}p1"
-    mkfs.f2fs -q "${LOOP_STORAGE_DEVICE}p2" && fsck "${LOOP_STORAGE_DEVICE}p2"
-    mkfs.f2fs -q "${LOOP_STORAGE_DEVICE}p3" && fsck "${LOOP_STORAGE_DEVICE}p3"
+    mkfs.ext4 -q "${LOOP_STORAGE_DEVICE}p1"
+    mkfs.f2fs -q "${LOOP_STORAGE_DEVICE}p2"
+    mkfs.f2fs -q "${LOOP_STORAGE_DEVICE}p3"
 
-    echo "Successfully created dummy storage device: '${LOOP_STORAGE_DEVICE}'."
+    if test_disk_integrity; then
+        echo "Successfully created dummy storage device: '${LOOP_STORAGE_DEVICE}'."
+        return 0
+    fi
+
```

i'd prefer it if you'd swap these two around.

Test the integrity, and exit on failure (fail early). If everything is okay, then return normally (without error).